numpy/_core/tests/test_memmap.py

```diff
@@ -207,10 +214,13 @@ def test_no_shape(self):
     def test_empty_array(self):
         # gh-12653
         with pytest.raises(ValueError, match='empty file'):
-            memmap(self.tmpfp, shape=(0,4), mode='w+')
+            memmap(self.tmpfp, shape=(0,4), mode='r')
 
-        self.tmpfp.write(b'\0')
+        # gh-27723
+        # empty memmap works with mode in ('w+','r+')
+        memmap(self.tmpfp, shape=(0,4), mode='w+')
 
+        self.tmpfp.write(b'\0')
```

Comment:
```suggestion
```
I think the `w+` call above already should have ensured the file is not empty.  (Also silly nit though).